crates/bevy_pbr/src/decal/clustered.rs

```diff
@@ -97,6 +98,72 @@ pub struct ClusteredDecal {
     pub tag: u32,
 }
 
+/// Cubemap layout defines the order of images in a packed cubemap image.
+#[derive(Default, Reflect, Debug, Clone, Copy)]
+pub enum CubemapLayout {
+    /// layout in a vertical cross format
+    ///    +y
+    /// -x -z +x
+    ///    -y
+    ///    +z
+    #[default]
+    CrossVertical = 0,
+    /// layout in a horizontal cross format
+    ///    +y
+    /// -x -z +x +z
+    ///    -y
+    CrossHorizontal = 1,
+    /// layout in a vertical sequence
+    ///   +x
+    ///   -y
+    ///   +y
+    ///   -y
+    ///   -z
+    ///   +z
+    SequenceVertical = 2,
+    /// layout in a horizontal sequence
+    /// +x -y +y -y -z +z
+    SequenceHorizontal = 3,
+}
+
+/// Add to a [`PointLight`] to add a light cookie effect.
+/// A light cookie is a texture mask applied to a light source to modulate its intensity,  
+/// simulating patterns like window shadows, gobo effects, or soft falloffs.
+#[derive(Clone, Component, Debug, Reflect)]
+#[reflect(Component, Debug)]
+#[require(PointLight)]
+pub struct PointLightCookie {
+    /// The cookie image. Only the R channel is read.
+    pub image: Handle<Image>,
+    /// The cubemap layout. The image should be a packed cubemap in one of the formats described by the [`CubemapLayout`] enum.
+    pub cubemap_layout: CubemapLayout,
+}
+
+/// Add to a [`SpotLight`] to add a light cookie effect.
+/// A light cookie is a texture mask applied to a light source to modulate its intensity,  
+/// simulating patterns like window shadows, gobo effects, or soft falloffs.
+#[derive(Clone, Component, Debug, Reflect)]
+#[reflect(Component, Debug)]
+#[require(SpotLight)]
+pub struct SpotLightCookie {
+    /// The cookie image. Only the R channel is read.
+    /// Note the border of the image should be entirely black to avoid leaking light.
+    pub image: Handle<Image>,
+}
+
+/// Add to a [`DirectionalLight`] to add a light cookie effect.
+/// A light cookie is a texture mask applied to a light source to modulate its intensity,  
+/// simulating patterns like window shadows, gobo effects, or soft falloffs.
+#[derive(Clone, Component, Debug, Reflect)]
+#[reflect(Component, Debug)]
+#[require(DirectionalLight)]
+pub struct DirectionalLightCookie {
+    /// The cookie image. Only the R channel is read.
+    pub image: Handle<Image>,
+    /// Whether to tile the image infinitely, or use only a single tile centered at the light's translation
+    pub tiled: bool,
+}
```

Comment:
I'm kind of surprised that directional light cookies exist given that I think of directional lights as infinitely large, but cool that you added this!

Reply:
There’s an option to tile or use a single centered instance of the texture; this was basically copied from the unity api

Reply:
I highly appreciate this addition, since it will probably make cloud shadows possible! Hopefully in the future this could be made to somehow work with a shader instead of just a static texture to add dynamism

Reply:
It should work already - you just need a camera rendering to the texture you use on the light.

Reply:
That's true, but I guess I was imagining something like what this [unity tutorial](https://youtu.be/JLXLUA_DIQc?si=dh7SVTYZ0c9Un4lu&t=497) does (timestamp relevant) where you can just put a full on material into the "light cookie" slot. But now that I think about this I am not sure how this would be done in bevy. Maybe Unity internally also just renders the material to the texture to facilitate this.
[On discord](https://discord.com/channels/691052431525675048/866787577687310356/1349784636829733049), @NthTensor  implied that with mode modular shader logic in bevy in the future, this could be possible, which would be exciting.